assignment01-voronoi/worldgen-voronoi.c

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+            else
+            {
+                gridMatrix[x][y] = storedSeeds[closestSeed].tile;
+            }
+        }
+    }
+}
+
+void generate_path(int leftX, int leftY, int botX, int botY, int rightX, int rightY, int topX, int topY)
+{
+
+    
+    int direction;
+
+    // pick one of the 3 "runs" for placing a pokemart
+    int PCplacement = 1 + rand() % 3;
+    int PCx = 0, PCy = 0;
+    int Mplacement = 1 + rand() %3;
+    int Mx = 0, My = 0;
+    // place pokemon center on left run to bottom gate
+    if (PCplacement == 1) {
+        if(leftX -1 != 0){
+        PCx = leftX - 1; // TODO: fixme
+        } else if (leftX +1 != 21){
+            PCx = leftX +1;
+        }
+        PCy = leftY + (rand() % (botY - leftY - 1));
+        printf("placement%d, PCx:%d, PCy:%d\n", PCplacement, PCx, PCy); 
+    }
+    //pokemart placement if 1
+    if(Mplacement == 1){
+        if( leftX - 1 != 0){
+            Mx = leftX -1;
+        } else if (leftX+1 != 21){
+            Mx = leftX +1;
+        }
+        Mx = leftY + (rand() % (botY - leftY -1));
+    }
+    
+    // left gate: go right until bottom gate
+    while (leftY < botY)
+    {
+        gridMatrix[leftX][leftY] = pathTile.ascii;
+        leftY++;
+    }
+
+    // place pokemon center on virtical run to right gate
+    if (PCplacement == 2) {
+        if((leftX-rightX) == 0) {
+            PCx = leftX;
+        }
+        else if( (leftX-rightX) < 0 ) {
+            PCx = leftX + (rand() % (rightX-leftX));
+        } 
+        else {
+            PCx = rightX + (rand() % (leftX-rightX));
+        }
+        PCy = leftY - 1;
+        printf("placement:%d, PCx:%d, PCy:%d\n", PCplacement, PCx, PCy); 
+    }
+    // place mart if 2
+    if(Mplacement == 2) {
+        if((leftX - rightX) == 0){
+            Mx = leftX;
+        }
+        else if ((leftX-rightX) < 0 ) {
+            Mx = leftX + (rand() % (rightX - leftX));
+        }
+        else {
+            Mx = rightX + (rand() % (leftX-rightX));
+        }
+        My = leftY -1;
+    }
+
+    direction = leftX > rightX ? -1 : 1;
+    // left gate: go in virtical direction of right gate
+    while (leftX != rightX)
+    {
+        gridMatrix[leftX][leftY] = pathTile.ascii;
+        leftX += direction;
+    }
+    
+    // place pokemon center on last run to right gate
+    if(PCplacement == 3) {
+        PCx = leftX - 1; // TODO: fixme
+        PCy = leftY + (rand() % (rightY - leftY + 1));
+        printf("placement:%d, PCx:%d, PCy:%d\n", PCplacement, PCx, PCy); 
+    }
+
+    if(Mplacement == 3) {
+        Mx = leftX -1;
+        My = leftY + (rand() % (rightY - leftY +1));
+    }
+
+    // left gate: go rest of way in horizontal to right gate
+    while (leftY != GRID_WIDTH)
+    {
+        gridMatrix[leftX][leftY] = pathTile.ascii;
+        leftY++;
+    }
+
+    // bottom gate: go up until right gate (leftX has been changed)
+    while (botX > leftX)
```

Comment:
so for example ur first "run" of the pokemart will use difference between botX and leftY and botY